SequencePhenotype.java

```diff
@@ -117,14 +119,35 @@ public double distance(Phenotype p) {
      *         here encoded more directly as risk of infection, which ranges from 0 to 1
      */
     public double riskOfInfection(Phenotype[] history) {
-        double risk = 1;
-        for (Phenotype sequence : history) {
-            SequencePhenotype seq = (SequencePhenotype) sequence;
-            if(seq.getSequence().length() == this.sequence.length()) {
-                risk *= this.distance(seq);
+        double fullImmuneRisk = 1.0;
+
+        int maxInput = this.sequence.length(); // The largest number of the range input.
+        int maxOutput = 1; // The largest number of the range output.
+
+        double slope = (double) (maxOutput) / (maxInput);
+
+        for(Phenotype pHistory : history) {
+            double inputDistance = this.distance(pHistory); // hamming distance of this and pHistory
+            double output = slope * (inputDistance);
+
+            double localImmuneRisk = 0.0;
+
+            switch (Parameters.crossImmunityFunction) {
+                case "linear":
+                    localImmuneRisk = output;
+                    break;
+                case "exponential":
+                    localImmuneRisk =  (1 - exp(-output));
+                    break;
+                case "exponentialSimplified":
+                    localImmuneRisk = 1 - exp(-inputDistance / Parameters.crossImmunityStrength);
```

Comment:
Did I have cross-immunity strength here and as an exponent below? Seems a little strange to do that.

Reply:
Oh, the exponent below was meant for the first two cases. Thanks for catching that!

(I tested the class with `Parameters.crossImmunityStrength: 1.0`, so it shouldn't have affected the shape of the tree.)